cmd/krel/cmd/release_notes.go

```diff
@@ -787,11 +899,393 @@ func (o *releaseNotesOptions) Validate() error {
 	}
 
 	// Options for PR creation
-	if releaseNotesOpts.createDraftPR || releaseNotesOpts.createWebsitePR {
-		if releaseNotesOpts.githubOrg == "" {
+	if o.createDraftPR || o.createWebsitePR {
+		if o.githubOrg == "" {
 			return errors.New("cannot generate the Release Notes PR without --org")
 		}
 	}
 
 	return nil
 }
+
+// Save the session to a file
+func (sd *sessionData) Save() error {
+	if sd.Date == 0 {
+		return errors.New("unable to save session, date is note defined")
+	}
+	if sd.Path == "" {
+		return errors.New("unable to save session, path is not defined")
+	}
+
+	jsonData, err := json.Marshal(sd)
+	if err != nil {
+		return errors.Wrap(err, "marshaling session data")
+	}
+
+	if err := ioutil.WriteFile(
+		fmt.Sprintf("%s%smaps-%d.json", sd.Path, string(os.PathSeparator), sd.Date),
+		jsonData, os.FileMode(0o644)); err != nil {
+		return errors.Wrap(err, "writing session data to disk")
+	}
+	return nil
+}
+
+// readFixSessions reads all the previous fixing data
+func readFixSessions(sessionPath string) (pullRequestChecklist map[int]bool, err error) {
+	files, err := ioutil.ReadDir(sessionPath)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading working directory")
+	}
+	pullRequestList := make([]int, 0)
+
+	// Look in the work dir for all json files
+	for _, fileData := range files {
+		currentSession := &sessionData{}
+		if strings.HasSuffix(fileData.Name(), ".json") {
+			jsonData, err := ioutil.ReadFile(
+				fmt.Sprintf("%s%s%s", sessionPath,
+					string(os.PathSeparator), fileData.Name()))
+			if err != nil {
+				return nil, errors.Wrapf(err, "reading session data from %s", fileData.Name())
+			}
+			if err := json.Unmarshal(jsonData, currentSession); err != nil {
+				return nil, errors.Wrapf(err, "unmarshalling session data in %s", fileData.Name())
+			}
+			pullRequestList = append(pullRequestList, currentSession.PullRequests...)
+		}
+	}
+
+	// Copy the PRs to a map for easy lookup
+	pullRequestChecklist = map[int]bool{}
+	for _, pr := range pullRequestList {
+		pullRequestChecklist[pr] = true
+	}
```

Comment:
What happens if a PR is mentioned multiple times? We'll overwrite the map with another one, and `ioutil.ReadDir` doesn't seem to have ordering guarantees. Could this lead to nondeterministic behaviour?

Reply:
We've maps override themselves in the order they come in. In the case of the krel flow it should not be a problem right now as no other map files should be in the directory, but it is definitely a good point to consider for the libraries in general. We should at least give them a predictable reading pattern.

Reply:
I guess we'd either want to order the files (so it's consistent at least), or do a map merge, or just error if there are multiples.